tests/recovery.test.ts

```diff
@@ -53,8 +65,46 @@ describe("recovery-tests", () => {
       await LocalRecovery.promise1;
       return ctxt.authenticatedUser;
     }
+
+    static recoveryCount = 0;
+    static readonly maxRecoveryAttempts = 5;
+    static deadLetterResolve: () => void;
+    static deadLetterPromise = new Promise<void>((resolve) => {
+      LocalRecovery.deadLetterResolve = resolve;
+    });
+
+
+    @Workflow({maxRecoveryAttempts: LocalRecovery.maxRecoveryAttempts})
+    static async deadLetterWorkflow(_ctxt: WorkflowContext) {
+      LocalRecovery.recoveryCount += 1
+      await LocalRecovery.deadLetterPromise;
+    }
   }
 
+  test("dead-letter-queue", async () => {
+    LocalRecovery.cnt = 0;
+    const dbosExec = (testRuntime as TestingRuntimeImpl).getDBOSExec();
+
+    const handle = await testRuntime.startWorkflow(LocalRecovery).deadLetterWorkflow();
+
+    for (let i = 0; i < LocalRecovery.maxRecoveryAttempts * 2; i++) {
+      await dbosExec.recoverPendingWorkflows();
+      expect(LocalRecovery.recoveryCount).toBeGreaterThanOrEqual(Math.min(i, LocalRecovery.maxRecoveryAttempts));
+      expect(LocalRecovery.recoveryCount).toBeLessThanOrEqual(LocalRecovery.maxRecoveryAttempts);
+    }
+
+    for (let i = 0; i < LocalRecovery.maxRecoveryAttempts * 2; i++) {
+      await testRuntime.startWorkflow(LocalRecovery, handle.getWorkflowUUID()).deadLetterWorkflow();
```

Comment:
Does it mean the workflow can still be started by user requests even if it's in the DLQ?

Reply:
Yes! That's deliberate, to allow recovery, we'll document it carefully.